synapse/http/endpoint.py

```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from twisted.internet.endpoints import SSL4ClientEndpoint, TCP4ClientEndpoint
+from twisted.internet.endpoints import HostnameEndpoint, wrapClientTLS
```

Comment:
`TCP4ClientEndpoint` is used as a default value in in the `SpiderEndpoint` and `SRVClientEndpoint` init functions.